### PR TITLE
[WNMGDS-2319] Storybook 7 upgrade

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

Upgrade to Storybook 7, resolve build issues around Typescript.

See the Storybook examples still work on the [demo url](https://cmsgov.github.io/design-system/branch/WNMGDS-2319/storybook-7-upgrade).

### CODE REVIEW

**Comment by forrestbaer** [comment]
Discovered one strange thing with the analytics menu:
image

**Comment by forrestbaer** [comment]
Getting 6 `test:unit` errors all relating to tooltip.

**Comment by forrestbaer** [comment]
Running into issues with VRT snapshots as well, probably because of the Docs pages, I'll see if I can work through that in this branch.

**Comment by forrestbaer** [comment]
I updated the VRT testing to work with the changes made and pushed the diffs it found, and hey, VRT interaction testing paying off! Take a look at `SingleInputDateField` with Picker and it no longer makes not-in-range items un-clickable, not sure how that happened. Maybe something with the story config

**Comment by pwolfert** [comment]
Looks like the current CI failures are TypeScript errors that might have to do with the react types upgrades

File: tests/browser/snapshots/components-text-field--multiline-field-cmsgov--firefox.png
Comment (zarahzachz) [inline tests/browser/snapshots/components-text-field--multiline-field-cmsgov--firefox.png:1]: 🤔 I wonder why the FF VRTs removed the resize handle from multi-line text inputs? Running it locally and I can clearly see the resizer (FF v 112)

File: tests/browser/snapshots/datefield--picker-interaction--cmsgov--Mobile-Chrome.png
Comment (zarahzachz) [inline tests/browser/snapshots/datefield--picker-interaction--cmsgov--Mobile-Chrome.png:1]: Logic appears broken for datepicker 😢  We have the `toDate` prop set to 2/15/23, which looks like it's supposed to make all following dates disabled (as prior VRT shows). Dates are now all accessible - don't know why a SB upgrade would impact this?

File: tests/browser/snapshots/components-text-field--multiline-field-cmsgov--firefox.png
Comment (forrestbaer) [reply tests/browser/snapshots/components-text-field--multiline-field-cmsgov--firefox.png:1]: It's there but it's tiny!

File: tests/browser/snapshots/datefield--picker-interaction--cmsgov--Mobile-Chrome.png
Comment (zarahzachz) [reply tests/browser/snapshots/datefield--picker-interaction--cmsgov--Mobile-Chrome.png:1]: I accidentally bumped the version of react-day-picker. This caused our local CSS to fall out of sync with the library and that's why things look broken.

Comment (forrestbaer) [reply tests/browser/snapshots/datefield--picker-interaction--cmsgov--Mobile-Chrome.png:1]: So glad it was easy! Was worried storybook was doing something strange.

File: package.json [hunk 1]
```diff
@@ -73,22 +73,23 @@
...
-    "@playwright/test": "^1.31.0",
-    "@storybook/addon-actions": "^6.5.16",
-    "@storybook/addon-essentials": "^6.5.16",
-    "@storybook/addon-links": "^6.5.16",
-    "@storybook/builder-webpack5": "^6.5.16",
-    "@storybook/manager-webpack5": "^6.5.16",
-    "@storybook/preset-scss": "^1.0.3",
-    "@storybook/react": "^6.5.16",
-    "@storybook/theming": "^6.5.16",
+    "@playwright/test": "^1.33.0",
```
Comment (zarahzachz) [inline package.json:85]: SB migration bumped this package and it's breaking tests in CI 😩 Addressing this issue now

File: .storybook/main.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  previewHead: (head) => `
+    ${head}
+    <link rel="stylesheet" type="text/css" title="themeCss" href="core-theme.css" />
+  `,
+  stories: [
+    '../packages/design-system/src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
+    '../packages/ds-healthcare-gov/src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
+    '../packages/ds-medicare-gov/src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
+  ],
+  staticDirs: ['./static/', '../packages/design-system/src'],
```
Comment (pwolfert) [inline .storybook/main.ts:17]: This actually clobbers a fix I recently made, which was removing the `'../packages/design-system/src'` item

**Review by pwolfert — CHANGES_REQUESTED** [review]

**Review by forrestbaer — APPROVED** [review]

**Review by pwolfert — APPROVED** [review]
LGTM 🎉

**Pull request merged**